### PR TITLE
Drill through fixes and cleanup

### LINKED ISSUE #10871: Drill down broken in v.0.33.2 when joining tables
**Describe the bug**
When summarizing a group of orders and grouping on a column from another joined table, the drill-down to "view these orders" breaks as it only looks for it in the main table
image

**Logs**
Log from metabase
image

No JS errors only warnings regarding mismatch of columns
image

**To Reproduce**
1. Go to  query builder
2. join two tables
3. summarize by count and group by column from the joined table
4. visualize results
5. click on count and select "view these orders"
6. See error

**Expected behavior**
I would expect to see those orders using the same query previously used

**Screenshots**
See above

**Information about your Metabase Installation:**
- Your browser and the version: Chrome
- Your operating system: Windows 10
- Your databases: MySQL 5.7
- Metabase version: 0.33.2
- Metabase hosting environment: Docker
- Metabase internal database: H2

**Severity**
I would say this is of high severity as it was working in version 0.32.10 and stopped working in the new Query builder, rendering the drill down options useless

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #10871

### CODE REVIEW

**Comment by tlrobinson** [comment]
@paulrosenzweig This should be ready for review

File: frontend/test/metabase/modes/components/actions/CountByTimeAction.unit.spec.js [hunk 1]
```diff
@@ -22,6 +22,6 @@ describe("CountByTimeAction", () => {
       aggregation: [["count"]],
       breakout: [["datetime-field", ["field-id", ORDERS.CREATED_AT.id], "day"]],
     });
-    expect(newCard.display).toEqual("bar");
+    expect(newCard.display).toEqual("line");
```
Comment (paulrosenzweig) [inline frontend/test/metabase/modes/components/actions/CountByTimeAction.unit.spec.js]: Why did this change?

**Review by paulrosenzweig — APPROVED** [review]
Looks good to me. I know this was a more general cleanup, but maybe it's worth adding some tests around the specific bug that prompted this? (drill in on joined fields)

File: frontend/test/metabase/modes/components/actions/CountByTimeAction.unit.spec.js (hunk 1 above)
Comment (tlrobinson) [reply frontend/test/metabase/modes/components/actions/CountByTimeAction.unit.spec.js]: I replaced our use of guessVisualization(card, tableMedata) with question.setDefaultDisplay(), which I guess had slightly different logic, but I think line is more correct for timeseries.

**Pull request merged**